tests/10-client-api.spec.js

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import dbHttpClient from '..';
+import isNode from 'detect-node';
+
+describe('http-client API', () => {
+  it('has proper exports', async () => {
+    should.exist(dbHttpClient);
+    dbHttpClient.should.have.keys([
+      'httpClient', 'ky', 'DEFAULT_HEADERS'
+    ]);
+    const {httpClient, ky} = dbHttpClient;
+    httpClient.should.be.a('function');
+    ky.should.be.a('function');
+  });
+  it('handles a network error', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      // hit a non-existent endpoint on the karma server to avoid CORS concerns
+      response = await httpClient.get('http://localhost:9876/does-not-exist');
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(response);
+    should.exist(err);
+    if(isNode) {
+      err.message.should.contain(
+        'request to http://localhost:9876/does-not-exist failed, reason: ' +
+        'connect ECONNREFUSED 127.0.0.1:9876');
+    } else {
+      err.message.should.contain('Not Found');
+    }
+  });
+  it('handles a get not found error', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      response = await httpClient.get('http://httpbin.org/status/404');
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(response);
+    should.exist(err);
+    err.message.should.contain('NOT FOUND');
+    should.exist(err.response);
+    should.exist(err.response.status);
+    err.response.status.should.equal(404);
+  });
+  it('should give a meaningful CORS error', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      response = await httpClient.get(
+        'https://example.com/55d2da26-b555-4950-8e12-fdab8de488a3');
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(response);
+    should.exist(err);
+    if(isNode) {
+      err.message.should.contain('Not Found');
+      should.exist(err.response);
+      should.exist(err.response.status);
+      err.response.status.should.equal(404);
+    } else { // failed to fetch may commonly be due to an issue with CORS
+      err.message.should.equal('Failed to fetch. Possible CORS error.');
+    }
+  });
+  it('succesfully makes request with default json headers', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      // TODO: use another site/mock for error response
```

Comment:
How important is it to fix these with a local server?  Should we make an issue for it or address it before first release?  httpbin sure is easier!

Reply:
httbin is awesome, we're rolling with it, removing the todo.